Tests/Fakes/FakeRequestPerformer.swift

```diff
@@ -2,55 +2,45 @@ import Foundation
 @testable import Swish
 import Result
 
-class FakeRequestPerformer: RequestPerformer {
-  let returnedJSON: [String: AnyObject]?
-  let statusCode: Int
-  var passedRequest: NSURLRequest?
-
-  init(returnedJSON: [String: AnyObject]? = .None, statusCode: Int = 200) {
-    self.returnedJSON = returnedJSON
-    self.statusCode = statusCode
-  }
+func serializeJSON(j: [String: AnyObject]) -> NSData? {
+  return try? NSJSONSerialization
+    .dataWithJSONObject(j, options: NSJSONWritingOptions(rawValue: 0))
+}
 
-  func performRequest(request: NSURLRequest, completionHandler: Result<HTTPResponse, NSError> -> Void) -> NSURLSessionDataTask {
-    passedRequest = request
-    let data = returnedJSON.flatMap {
-      try? NSJSONSerialization.dataWithJSONObject($0, options: NSJSONWritingOptions(rawValue: 0))
-    }
+enum ResponseData {
+  case Data(NSData?)
```

Comment:
I don't think that matters, since we're just passing this through. The rest of the API deals with optional `NSData`, so it feels nice (to me) to have this be explicit.